### PR TITLE
Remove redundant class branch

### PR DESCRIPTION
**Pull request by JrmCkh**
1. Removed redundant classes (e.g., Person)
2. Converted classes that has "AddressBook" to "Scheduler"
3. Changed all comments includes address book to scheduler

### CODE REVIEW

File: src/test/java/seedu/address/testutil/AddressBookBuilder.java
Comment (lhy-hoyin) [inline src/test/java/seedu/address/testutil/AddressBookBuilder.java:1]: variable name and parameter name (e.g line19)

File: src/test/java/seedu/address/testutil/TypicalPersons.java [hunk 1]
```diff
@@ -62,8 +62,8 @@ private TypicalPersons() {} // prevents instantiation
     /**
      * Returns an {@code AddressBook} with all the typical persons.
      */
-    public static AddressBook getTypicalAddressBook() {
-        AddressBook ab = new AddressBook();
+    public static Scheduler getTypicalAddressBook() {
+        Scheduler ab = new Scheduler();
```
Comment (lhy-hoyin) [inline src/test/java/seedu/address/testutil/TypicalPersons.java:66]: Variable name
`ab` was meant for AddressBook

File: src/test/java/seedu/address/testutil/TypicalPersons.java [hunk 2]
```diff
@@ -62,8 +62,8 @@ private TypicalPersons() {} // prevents instantiation
     /**
      * Returns an {@code AddressBook} with all the typical persons.
      */
-    public static AddressBook getTypicalAddressBook() {
-        AddressBook ab = new AddressBook();
+    public static Scheduler getTypicalAddressBook() {
```
Comment (lhy-hoyin) [inline src/test/java/seedu/address/testutil/TypicalPersons.java:65]: method name

**Review by lhy-hoyin — CHANGES_REQUESTED** [review]
Generally a good job on the major refactor. But it seems that the tests need a bit more refactoring. Do let me know if you need help with it :D

File: src/main/java/seedu/address/commons/exceptions/IllegalValueException.java [hunk 3]
```diff
@@ -13,7 +13,7 @@ public IllegalValueException(String message) {
 
     /**
      * @param message should contain relevant information on the failed constraint(s)
-     * @param cause of the main exception
+     * @param cause   of the main exception
```
[the remark this answers is not in the document]
Comment (JrmCkh) [reply src/main/java/seedu/address/commons/exceptions/IllegalValueException.java:16]: This was caused by the "auto reformat code" of intellij, which automaticallys lines up the message in the same line

**Review by lhy-hoyin — APPROVED** [review]
Code in tp.Main module looks good.
Changes in tp.Test will need to be worked on in the future.

File: src/test/java/seedu/address/testutil/TypicalPersons.java (hunk 1 above)
Comment (JrmCkh) [reply src/test/java/seedu/address/testutil/TypicalPersons.java:66]: Will be doing next

File: src/test/java/seedu/address/testutil/TypicalPersons.java (hunk 2 above)
Comment (JrmCkh) [reply src/test/java/seedu/address/testutil/TypicalPersons.java:65]: Will be doing next

File: src/test/java/seedu/address/storage/StorageManagerTest.java [hunk 4]
```diff
@@ -54,10 +54,10 @@ public void addressBookReadSave() throws Exception {
          * {@link JsonAddressBookStorage} class.
          * More extensive testing of UserPref saving/reading is done in {@link JsonAddressBookStorageTest} class.
          */
-        AddressBook original = getTypicalAddressBook();
+        Scheduler original = getTypicalAddressBook();
         storageManager.saveAddressBook(original);
-        ReadOnlyAddressBook retrieved = storageManager.readAddressBook().get();
-        assertEquals(original, new AddressBook(retrieved));
+        ReadOnlyScheduler retrieved = storageManager.readAddressBook().get();
```
[the remark this answers is not in the document]
Comment (JrmCkh) [reply src/test/java/seedu/address/storage/StorageManagerTest.java:59]: Will be doing next

File: src/test/java/seedu/address/storage/StorageManagerTest.java [hunk 5]
```diff
@@ -54,10 +54,10 @@ public void addressBookReadSave() throws Exception {
          * {@link JsonAddressBookStorage} class.
          * More extensive testing of UserPref saving/reading is done in {@link JsonAddressBookStorageTest} class.
          */
-        AddressBook original = getTypicalAddressBook();
+        Scheduler original = getTypicalAddressBook();
```
[the remark this answers is not in the document]
Comment (JrmCkh) [reply src/test/java/seedu/address/storage/StorageManagerTest.java:57]: Will be doing next

File: src/test/java/seedu/address/storage/StorageManagerTest.java [hunk 6]
```diff
@@ -24,7 +24,7 @@ public class StorageManagerTest {
 
     @BeforeEach
     public void setUp() {
-        JsonAddressBookStorage addressBookStorage = new JsonAddressBookStorage(getTempFilePath("ab"));
+        JsonSchedulerStorage addressBookStorage = new JsonSchedulerStorage(getTempFilePath("ab"));
```
[the remark this answers is not in the document]
Comment (JrmCkh) [reply src/test/java/seedu/address/storage/StorageManagerTest.java:27]: Will be doing next

File: src/test/java/seedu/address/storage/JsonSerializableAddressBookTest.java [hunk 7]
```diff
@@ -24,8 +24,8 @@ public class JsonSerializableAddressBookTest {
     public void toModelType_typicalPersonsFile_success() throws Exception {
         JsonSerializableScheduler dataFromFile = JsonUtil.readJsonFile(TYPICAL_PERSONS_FILE,
                 JsonSerializableScheduler.class).get();
-        AddressBook addressBookFromFile = dataFromFile.toModelType();
-        AddressBook typicalPersonsAddressBook = TypicalPersons.getTypicalAddressBook();
+        Scheduler addressBookFromFile = dataFromFile.toModelType();
+        Scheduler typicalPersonsAddressBook = TypicalPersons.getTypicalAddressBook();
```
[the remark this answers is not in the document]
Comment (JrmCkh) [reply src/test/java/seedu/address/storage/JsonSerializableAddressBookTest.java:28]: Will be doing next

File: src/test/java/seedu/address/logic/commands/CommandTestUtil.java [hunk 8]
```diff
@@ -106,7 +106,7 @@ public static void assertCommandSuccess(Command command, Model actualModel, Stri
     public static void assertCommandFailure(Command command, Model actualModel, String expectedMessage) {
         // we are unable to defensively copy the model for comparison later, so we can
         // only do so by copying its components.
-        AddressBook expectedAddressBook = new AddressBook(actualModel.getAddressBook());
+        Scheduler expectedAddressBook = new Scheduler(actualModel.getAddressBook());
```
[the remark this answers is not in the document]
Comment (JrmCkh) [reply src/test/java/seedu/address/logic/commands/CommandTestUtil.java:109]: Will be doing next

File: src/test/java/seedu/address/logic/commands/ClearCommandTest.java [hunk 9]
```diff
@@ -24,7 +24,7 @@ public void execute_emptyAddressBook_success() {
     public void execute_nonEmptyAddressBook_success() {
         Model model = new ModelManager(getTypicalAddressBook(), new UserPrefs());
         Model expectedModel = new ModelManager(getTypicalAddressBook(), new UserPrefs());
```
[the remark this answers is not in the document]
Comment (JrmCkh) [reply src/test/java/seedu/address/logic/commands/ClearCommandTest.java:26]: Will be doing next

File: src/test/java/seedu/address/logic/commands/AddCommandTest.java [hunk 10]
```diff
@@ -186,8 +186,8 @@ public void addPerson(Person person) {
         }
 
         @Override
-        public ReadOnlyAddressBook getAddressBook() {
-            return new AddressBook();
+        public ReadOnlyScheduler getAddressBook() {
```
[the remark this answers is not in the document]
Comment (JrmCkh) [reply src/test/java/seedu/address/logic/commands/AddCommandTest.java:189]: Will be doing next

File: src/test/java/seedu/address/logic/commands/AddCommandTest.java [hunk 11]
```diff
@@ -114,12 +114,12 @@ public void addPerson(Person person) {
...
 
         @Override
-        public void setAddressBook(ReadOnlyAddressBook newData) {
+        public void setAddressBook(ReadOnlyScheduler newData) {
             throw new AssertionError("This method should not be called.");
         }
 
         @Override
-        public ReadOnlyAddressBook getAddressBook() {
+        public ReadOnlyScheduler getAddressBook() {
```
[the remark this answers is not in the document]
Comment (JrmCkh) [reply src/test/java/seedu/address/logic/commands/AddCommandTest.java:122]: Will be doing next

File: src/test/java/seedu/address/logic/LogicManagerTest.java [hunk 12]
```diff
@@ -148,13 +148,13 @@ private void assertCommandFailure(String inputCommand, Class<? extends Throwable
...
-    private static class JsonAddressBookIoExceptionThrowingStub extends JsonAddressBookStorage {
-        private JsonAddressBookIoExceptionThrowingStub(Path filePath) {
+    private static class JsonSchedulerIoExceptionThrowingStub extends JsonSchedulerStorage {
+        private JsonSchedulerIoExceptionThrowingStub(Path filePath) {
             super(filePath);
         }
 
         @Override
-        public void saveAddressBook(ReadOnlyAddressBook addressBook, Path filePath) throws IOException {
+        public void saveAddressBook(ReadOnlyScheduler addressBook, Path filePath) throws IOException {
```
[the remark this answers is not in the document]
Comment (JrmCkh) [reply src/test/java/seedu/address/logic/LogicManagerTest.java:157]: Will be doing next

File: src/test/java/seedu/address/logic/LogicManagerTest.java [hunk 13]
```diff
@@ -70,8 +70,8 @@ public void execute_validCommand_success() throws Exception {
     @Test
     public void execute_storageThrowsIoException_throwsCommandException() {
         // Setup LogicManager with JsonAddressBookIoExceptionThrowingStub
-        JsonAddressBookStorage addressBookStorage =
-                new JsonAddressBookIoExceptionThrowingStub(temporaryFolder.resolve("ioExceptionAddressBook.json"));
+        JsonSchedulerStorage addressBookStorage =
+                new JsonSchedulerIoExceptionThrowingStub(temporaryFolder.resolve("ioExceptionAddressBook.json"));
```
[the remark this answers is not in the document]
Comment (JrmCkh) [reply src/test/java/seedu/address/logic/LogicManagerTest.java:74]: Will be doing next

File: src/test/java/seedu/address/logic/LogicManagerTest.java [hunk 14]
```diff
@@ -42,8 +42,8 @@ public class LogicManagerTest {
 
     @BeforeEach
     public void setUp() {
-        JsonAddressBookStorage addressBookStorage =
-                new JsonAddressBookStorage(temporaryFolder.resolve("addressBook.json"));
+        JsonSchedulerStorage addressBookStorage =
+                new JsonSchedulerStorage(temporaryFolder.resolve("addressBook.json"));
```
[the remark this answers is not in the document]
Comment (JrmCkh) [reply src/test/java/seedu/address/logic/LogicManagerTest.java:46]: Will be doing next

File: src/test/java/seedu/address/testutil/AddressBookBuilder.java
Comment (JrmCkh) [reply src/test/java/seedu/address/testutil/AddressBookBuilder.java:1]: Will be doing next

File: src/test/java/seedu/address/logic/parser/AddressBookParserTest.java
[the remark this answers is not in the document]
Comment (JrmCkh) [reply src/test/java/seedu/address/logic/parser/AddressBookParserTest.java:1]: Will be doing next

File: src/test/java/seedu/address/model/AddressBookTest.java
[the remark this answers is not in the document]
Comment (JrmCkh) [reply src/test/java/seedu/address/model/AddressBookTest.java:1]: Will be doing next

File: src/test/java/seedu/address/storage/JsonSchedulerStorageTest.java
[the remark this answers is not in the document]
Comment (JrmCkh) [reply src/test/java/seedu/address/storage/JsonSchedulerStorageTest.java:1]: Will be doing next

**Pull request merged**